src/uct/ugni/smsg/ugni_smsg_iface.c

```diff
@@ -0,0 +1,460 @@
+/**
+ * Copyright (c) UT-Battelle, LLC. 2014-2015. ALL RIGHTS RESERVED.
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2014.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include <pmi.h>
+#include "ucs/type/class.h"
+
+#include <ucs/arch/cpu.h>
+#include <uct/ugni/base/ugni_iface.h>
+#include "ugni_smsg_iface.h"
+#include "ugni_smsg_ep.h"
+
+#define UCT_UGNI_SMSG_TL_NAME "ugni_smsg"
+
+static ucs_config_field_t uct_ugni_smsg_iface_config_table[] = {
+    {"", "ALLOC=huge,mmap,heap", NULL,
+     ucs_offsetof(uct_ugni_iface_config_t, super),
+     UCS_CONFIG_TYPE_TABLE(uct_iface_config_table)},
+
+    UCT_IFACE_MPOOL_CONFIG_FIELDS("SMSG", -1, 0, "smsg",
+                                  ucs_offsetof(uct_ugni_iface_config_t, mpool),
+                                  "\nAttention: Setting this param with value != -1 is a dangerous thing\n"
+                                  "and could cause deadlock or performance degradation."),
+
+    {NULL}
+};
+
+static void progress_local_cq(uct_ugni_smsg_iface_t *iface){
+    gni_return_t ugni_rc;
+    gni_cq_entry_t event_data;
+    uct_ugni_smsg_desc_t message_data;
+    uct_ugni_smsg_desc_t *message_pointer;
+
+    if(0 == iface->super.outstanding){
+        return;
+    }
+
+    ugni_rc = GNI_CqGetEvent(iface->super.local_cq, &event_data);
+    if(GNI_RC_NOT_DONE == ugni_rc){
```

Comment:
@MattBBaker please grep for "==" in the diff and revert the order to CONST == VAR. 10x!